tests/container_tests/test_zip_container.py

```diff
@@ -559,3 +560,258 @@ def func():
+        # |   | - testfile3
+        # | - testfile4
+
+        self.test_string = "this is a test string\n"
+        self.fs_handler = chainerio.create_handler("posix")
+
+        # the most outside zip
+        self.zip_file_name = "outside.zip"
+
+        # nested zip and nested file
+        self.tmpdir = tempfile.TemporaryDirectory()
+
+        # directory and file
+        self.dir1_name = NO_DIRECTORY_FILENAME_LIST["dir1_name"]
+        self.dir2_name = NO_DIRECTORY_FILENAME_LIST["dir2_name"]
+        self.dir3_name = NO_DIRECTORY_FILENAME_LIST["dir3_name"]
+        self.testfile1_name = NO_DIRECTORY_FILENAME_LIST["testfile1_name"]
+        self.testfile2_name = NO_DIRECTORY_FILENAME_LIST["testfile2_name"]
+        self.testfile3_name = NO_DIRECTORY_FILENAME_LIST["testfile3_name"]
+        self.testfile4_name = NO_DIRECTORY_FILENAME_LIST["testfile4_name"]
+
+        # paths used in making outside.zip
+        dir1_path = os.path.join(self.tmpdir.name, self.dir1_name)
+        dir2_path = os.path.join(dir1_path, self.dir2_name)
+        dir3_path = os.path.join(self.tmpdir.name, self.dir3_name)
+        testfile1_path = os.path.join(dir1_path, self.testfile1_name)
+        testfile2_path = os.path.join(dir2_path, self.testfile2_name)
+        testfile3_path = os.path.join(dir3_path, self.testfile3_name)
+        testfile4_path = os.path.join(self.tmpdir.name, self.testfile4_name)
+
+        # paths used in tests
+        for dir in [dir1_path, dir2_path, dir3_path]:
+            os.mkdir(dir)
+
+        for file_path in [testfile1_path, testfile2_path,
+                          testfile3_path, testfile4_path]:
+            with open(file_path, "w") as f:
+                f.write(self.test_string)
+
+        # create zip without directory
+        self.pwd = os.getcwd()
+        os.chdir(self.tmpdir.name)
+        cmd = ["zip", "-rD", self.zip_file_name, "."]
+
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        assert stderr == b""
+
+    def tearDown(self):
+        os.chdir(self.pwd)
+        self.tmpdir.cleanup()
+
+    @parameterized.expand([
+        # default case get the first level from the root
+        ["", [NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+              NO_DIRECTORY_FILENAME_LIST["dir3_name"],
+              NO_DIRECTORY_FILENAME_LIST["testfile4_name"]],
+         False],
+        # Problem 1 in issue #66
+        [NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+         [NO_DIRECTORY_FILENAME_LIST["testfile1_name"],
+          NO_DIRECTORY_FILENAME_LIST["dir2_name"]],
+         False],
+        # problem 2 in issue #66
+        [os.path.join(NO_DIRECTORY_FILENAME_LIST["dir1_name"],
+                      NO_DIRECTORY_FILENAME_LIST["dir2_name"]),
+         [NO_DIRECTORY_FILENAME_LIST["testfile2_name"]],
+         False],
+        # not normalized path
+        ['testdir1//testfile//../',
```

Comment:
Why not use `os.path.join()` and `NO_DIRECTORY_FILENAME_LIST`?

Reply:
If '//' is important, what about this?
`f'{NO_DIRECTORY_FILENAME_LIST["dir1_name"]}//{NO_DIRECTORY_FILENAME_LIST["dir2_name"]}//../'`

Reply:
Sorry, f string is not supported in `python 3.5.2`. I will use the format instead
https://stackoverflow.com/questions/42126794/python-3-returns-invalid-syntax-when-trying-to-perform-string-interpolation

Reply:
https://docs.python.org/3/whatsnew/3.6.html#pep-498-formatted-string-literals